workers/loc.api/sync/data.inserter/api.middleware.handler.after.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const { searchClosePrice } = require('./halpers')
+
+class ApiMiddlewareHandlerAfter {
+  constructor (reportService, dao) {
+    this.reportService = reportService
+    this.dao = dao
+  }
+
+  async _getPositionsHistory ({ auth }, apiRes, isCheckCall) {
+    if (isCheckCall) {
+      return apiRes
+    }
+
+    const res = []
+
+    for (const position of apiRes.res) {
+      const {
+        basePrice,
+        symbol,
+        mtsUpdate: end,
+        id
+      } = position
+
+      if (
+        !symbol ||
+        typeof symbol !== 'string' ||
+        /tBFX/gi.test(symbol) ||
+        !Number.isInteger(end) ||
+        !Number.isInteger(id) ||
+        !Number.isFinite(basePrice)
+      ) {
+        res.push({
+          ...position,
+          closePrice: null
+        })
+
+        continue
+      }
+
+      const closePrice = await searchClosePrice(
+        this.dao,
+        {
+          auth,
+          symbol,
+          end,
+          id
+        }
+      )
+
+      if (!Number.isFinite(closePrice)) {
+        res.push({
+          ...position,
+          closePrice,
+          pl: null,
+          plPerc: null
+        })
+
+        continue
+      }
+
+      const pl = closePrice - basePrice
```

Comment:
const pl = (closePrice - basePrice)  * traded amount  // Let me double check this

Reply:
It calculates as you propose

![screenshot from 2019-02-13 10-01-20](https://user-images.githubusercontent.com/16489235/52696035-a3aa3c80-2f76-11e9-9780-401821dbd239.png)

Let me know if it need to change

Reply:
yes, but this is a tricky case as the amount is 1, but if amount was 0.7 would be (closePrice - basePrice) * 0.7